src/node/audio_buffer_source.rs

```diff
@@ -1461,14 +1590,25 @@ mod tests {
         src.set_loop(true);
         src.set_loop_start(0.5); // outside of buffer duration
         src.set_loop_end(1.5); // outside of buffer duration
-
         src.start();
 
         let result = context.start_rendering_sync(); // should terminate
         let channel = result.get_channel_data(0);
 
-        assert_float_eq!(channel[0], 1.0, abs_all <= 0.);
-        assert_float_eq!(channel[1..], [0.; 48_000 - 1][..], abs_all <= 0.);
+        // Both loop points will be clamped to buffer duration due to rules defined at
+        // https://webaudio.github.io/web-audio-api/#dom-audiobuffersourcenode-loopstart
+        // https://webaudio.github.io/web-audio-api/#dom-audiobuffersourcenode-loopend
+        // Thus it violates the rule defined in
+        // https://webaudio.github.io/web-audio-api/#playback-AudioBufferSourceNode
+        // `loopStart >= 0 && loopEnd > 0 && loopStart < loopEnd`
+        // Hence the whole buffer should be looped
```

Comment:
Yes, I agree. I did not put too much thought into the expected result earlier, just focused on fixing the indefinite loop. Thanks!